docs/proposals/Ashish_Mahajan_Versioning_Proposal.md

```diff
@@ -0,0 +1,241 @@
+# Proposal : Branch-Based Version Management System for DISCOVER Cookbook
+
+## Abstract
+
+This proposal outlines a comprehensive implementation strategy for a branch-based version management system for the DISCOVER Cookbook following industry-standard documentation versioning practices as used by scikit-learn, Matplotlib, NetworkX, and other NumFOCUS projects. The implementation will enable users to seamlessly navigate between different cookbook editions (v2.0/main and v1.0/master) through a version selector integrated with the existing Jupyter Book interface, while maintaining the current GitHub Pages infrastructure.
+
+## Detailed description
+
+### Current Architecture Analysis
+
+The DISCOVER Cookbook currently maintains two separate versions with no navigation between them:
+
+- Current version (v2.0) in the main branch, deployed at https://discover-cookbook.numfocus.org
```

Comment:
No there is just the main branch. The master branch is just an old stale branch from when git changed the default branch name.